pkg/skaffold/deploy/helm.go

```diff
@@ -512,6 +523,38 @@ func (h *HelmDeployer) Render(context.Context, io.Writer, []build.Artifact, []La
 	return errors.New("not yet implemented")
 }
 
+// binVer returns the version of the helm binary found in PATH. May be cached.
+func (h *HelmDeployer) binVer(ctx context.Context) (semver.Version, error) {
+	// Return the cached version value if non-zero
+	if h.bV.Major != 0 && h.bV.Minor != 0 {
+		return h.bV, nil
+	}
+
+	var b bytes.Buffer
+	if err := h.helm(ctx, &b, false, "version", "--short", "-c"); err != nil {
+		return semver.Version{}, errors.Wrap(err, "helm version")
+	}
+	bs := b.Bytes()
+
+	// raw for 3.1: "v3.1.0+gb29d20b"
+	// raw for 2.15: "Client: v2.15.1+gcf1de4f"
+	raw := string(bs)
+	idx := strings.Index(raw, "v")
```

Comment:
Please add tests for the L542 to L551 for different helm version strings.

Reply:
Will do. Until the follow-up PR is submitted, this will unfortunately will mean that we have to test a private function, as this change is advisory only, and has no side-effects to the public functions.

Reply:
This now has unit tests.